src/dataflow/src/source/kafka.rs

```diff
@@ -762,6 +761,16 @@ impl ConsumerContext for GlueConsumerContext {
     }
 }
 
-fn has_partition(worker_id: i32, worker_count: i32, partition_id: i32) -> bool {
-    (partition_id % worker_count) == worker_id
+// Calculate whether or not the given worker should be responsible for reading from the given
+// source and partition
+fn has_partition(
+    worker_id: i32,
+    worker_count: i32,
+    source_id: SourceInstanceId,
+    partition_id: i32,
+) -> bool {
+    // Distribute read responsibility across workers
+    use differential_dataflow::hashable::Hashable;
+    let hash = source_id.hashed() as i32;
```

Comment:
Not relevant here, but someone should refactor all this code to not use `i32`s. These `as` conversions everywhere stress me out. `as` is the most misdesigned part of Rust IMO.

Reply:
I filed https://github.com/MaterializeInc/materialize/issues/5105 about this.

Reply:
Thanks for explaining why these variables were i32s. Agreed that we try to avoid casting where possible.